Form/Type/CategoryType.php

```diff
@@ -1,66 +1,89 @@
 <?php
+
 /*
  * This file is part of EC-CUBE
  *
- * Copyright(c) 2016 LOCKON CO.,LTD. All Rights Reserved.
+ * Copyright(c) LOCKON CO.,LTD. All Rights Reserved.
+ *
  * http://www.lockon.co.jp/
  *
  * For the full copyright and license information, please view the LICENSE
  * file that was distributed with this source code.
-*/
+ */
 
 namespace Plugin\ProductPriority\Form\Type;
 
-use Eccube\Application;
 use Eccube\Entity\Category;
-use Plugin\ProductPriority\Constant;
+use Eccube\Repository\CategoryRepository;
+use Plugin\ProductPriority\Entity\ProductPriority;
+use Plugin\ProductPriority\Repository\ProductPriorityRepository;
+use Symfony\Bridge\Doctrine\Form\Type\EntityType;
 use Symfony\Component\Form\AbstractType;
 use Symfony\Component\Form\FormBuilderInterface;
 
 class CategoryType extends AbstractType
 {
-    protected $app;
+    /**
+     * @var CategoryRepository
+     */
+    private $categoryRepository;
 
-    public function __construct(Application $app)
+    /**
+     * @var ProductPriorityRepository
+     */
+    private $productPriorityRepository;
+
+    /**
+     * CategoryType constructor.
+     *
+     * @param CategoryRepository $categoryRepository
+     * @param ProductPriorityRepository $productPriorityRepository
+     */
+    public function __construct(CategoryRepository $categoryRepository, ProductPriorityRepository $productPriorityRepository)
     {
-        $this->app = $app;
+        $this->categoryRepository = $categoryRepository;
+        $this->productPriorityRepository = $productPriorityRepository;
     }
 
+    /**
+     * @param FormBuilderInterface $builder
+     * @param array $options
+     */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
-        $Categories = $this->app['eccube.repository.category']
-            ->getList(null, true);
-
-        $countByCategory = $this->app['eccube.plugin.product_priority.repository.product_priority']
-            ->getPriorityCountGroupByCategory();
+        $Categories = $this->categoryRepository->getList(null, true);
+        $countByCategory = $this->productPriorityRepository->getPriorityCountGroupByCategory();
 
         $emptyValue = sprintf(
-            '全ての商品(%s)',
-            isset($countByCategory[Constant::CATEGORY_ID_ALL_PRODUCT])
-                ? $countByCategory[Constant::CATEGORY_ID_ALL_PRODUCT]
+            trans('product_priority.form.category.empty'),
+            isset($countByCategory[ProductPriority::CATEGORY_ID_ALL_PRODUCT])
+                ? $countByCategory[ProductPriority::CATEGORY_ID_ALL_PRODUCT]
                 : 0
```

Comment:
@lqdung-lockon should use placeholder when translate instead of `sprintf`
```
trans('product_priority.form.category.empty', ['%placeholder%' =>isset($countByCategory[ProductPriority::CATEGORY_ID_ALL_PRODUCT]) ?  $countByCategory[ProductPriority::CATEGORY_ID_ALL_PRODUCT] : 0])
```